Automattic/wp-calypso
# format-currency: Refactor to use createFormatter factory

## Proposed Changes

This PR refactors how the `@automattic/format-currency` library is organized. Previously it was a set of standalone functions (mainly the `formatCurrency()` default export). However, over time it has gained some persistent state, like the addition of `setDefaultLocale()`. That state has been stored globally. As the package is likely to gain more persistent state over time (eg: https://github.com/Automattic/wp-calypso/pull/74351 which allows setting currency overrides) and global state makes things like testing somewhat difficult, this PR refactors the package to move all the formatting functions into methods of a "CurrencyFormatter" object created by the `createFormatter()` factory function.

In order to preserve backwards compatibility, this PR adds a default global `CurrencyFormatter` and exports aliases of each of its methods so they will continue to work as before. However, it gives more control to any code that wants to use a fresh formatter without influencing the global state.

## Testing Instructions

Automated tests should cover everything.

View a page that has prices (eg: `/plans` or `/checkout`) and verify that the prices are still shown.
